src/ros_vrpn_client.cpp

```diff
@@ -29,103 +29,125 @@
-    //this->tracker->register_change_handler(NULL, track_target_acceleration);
-
-    tracker->print_latest_report();
-  }
+class Rigid_Body {
 
-  void publish_target_state(TargetState *target_state)
-  {
-    br.sendTransform(target_state->target);
-    target_pub.publish(target_state->target);
-  }
+  public:
+    // Constructor
+    Rigid_Body(ros::NodeHandle& nh, std::string server_ip, int port, const std::string& object_name)
+    {
+      // Advertising published topics.
+      measured_target_transform_publisher = nh.advertise<geometry_msgs::TransformStamped>("raw_vicon", 10);
+      estimated_target_transform_publisher = nh.advertise<geometry_msgs::TransformStamped>("pose", 10);
+      estimated_target_odometry_publisher = nh.advertise<nav_msgs::Odometry>("odometry", 10);
+      // Connecting to the vprn device and creating an associated tracker.
+      std::stringstream connection_name;
+      connection_name << server_ip << ":" << port;
+      connection = vrpn_get_connection_by_name(connection_name.str().c_str());
+      tracker = new vrpn_Tracker_Remote(object_name.c_str(), connection);
+      tracker->print_latest_report();
+      // Registering a callback to be called when a new data is made available by the vrpn sever.
+      this->tracker->register_change_handler(NULL, track_target);
+      tracker->print_latest_report();
+      // NOTE(millanea@ethz.ch): The following callbacks should be added if they're available.
+      //                         See detailed note above.
+      //this->tracker->register_change_handler(NULL, track_target_velocity);
+      //this->tracker->register_change_handler(NULL, track_target_acceleration);
+    }
 
-  void publish_odometry(TargetState *target_state)
-  {
-    odometry_pub.publish(target_state->odometry);
-  }
+    // Publishes the raw measured target state to the transform message.
+    void publish_measured_transform(TargetState *target_state)
+    {
+      measured_target_transform_publisher.publish(target_state->measured_transform);
+    }
 
-  void step_vrpn()
-  {
-    this->tracker->mainloop();
-    this->connection->mainloop();
-  }
+    // Publishes the estimated target state to the transform message and sends tranform.
+    void publish_estimated_transform(TargetState *target_state)
+    {
+      br.sendTransform(target_state->estimated_transform);
+      estimated_target_transform_publisher.publish(target_state->estimated_transform);
+    }
+
+    // Publishes the estimated target state to the odometry message.
+    void publish_estimated_odometry(TargetState *target_state)
+    {
+      estimated_target_odometry_publisher.publish(target_state->estimated_odometry);
+    }
+
+    // Passes contol to the vrpn client.
+    void step_vrpn()
+    {
+      this->tracker->mainloop();
+      this->connection->mainloop();
+    }
+
+  private:
+    // Publishers
+    ros::Publisher measured_target_transform_publisher;
```

Comment:
Trailing _ after member names in a class. pub_ is also okay abbreviation to save space, if you prefer. Other than that, these names are a big improvement, thanks! :)